plugins/UsersManager/Model.php

```diff
@@ -722,6 +755,35 @@ public function addUserAccess($userLogin, $access, $idSites)
         }
     }
 
+    public function updateUserAccessConditionally(string $userLogin, int $idSite, string $newAccess, string $previousAccess): bool
+    {
+        $db = $this->getDb();
+
+        $updateSql = "
+            UPDATE " . Common::prefixTable("access") . "
+            SET access = ?
+            WHERE login = ?
+              AND idsite = ?
+              AND access = ?
+        ";
+
+        $db->query($updateSql, [$newAccess, $userLogin, $idSite, $previousAccess]);
+
+        // Performing query again because rows affected won't change when $newAccess == $previousAccess
+        $selectSql = "
+            SELECT COUNT(*) FROM " . Common::prefixTable("access") . "
+            WHERE login = ?
+              AND idsite = ?
+              AND access = ?
+        ";
+
+        if ($db->fetchOne($selectSql, [$userLogin, $idSite, $newAccess]) === 0) {
```

Comment:
Depending on the use database adapter it might return a string or an integer. Therefor a casting might be good.
```suggestion
        if ((int)$db->fetchOne($selectSql, [$userLogin, $idSite, $newAccess]) === 0) {
```